packages/medusa/src/types/global.ts

```diff
@@ -51,6 +55,19 @@ export type ModuleDefinition = {
   isRequired?: boolean
 }
 
+export type LoaderOptions = {
+  container: MedusaContainer
+  configModule: ConfigModule
+  logger?: Logger
+}
+
+export type Constructor<T> = new (...args: any[]) => T
```

Comment:
todo: Update the existing one above `ClassConstructor`, I have testing to build and it does not impact negatively the rest of the code base